src/momento/simple_cache_client.py

```diff
@@ -11,38 +12,35 @@
     DeleteCacheResponse,
     ListCachesResponse,
 )
+from .config.configuration import Configuration
 from .internal.synchronous._scs_control_client import _ScsControlClient
 from .internal.synchronous._scs_data_client import _ScsDataClient
 
 
 class SimpleCacheClient:
     def __init__(
         self,
-        auth_token: str,
-        default_ttl_seconds: int,
-        request_timeout_ms: Optional[int] = None,
+        configuration: Configuration,
+        credential_provider: CredentialProvider,
+        default_ttl: timedelta,
     ):
         """Creates an async SimpleCacheClient
 
         Args:
-            auth_token (str): Momento Token to authenticate the requests with Simple Cache Service
-            default_ttl_seconds (int): A default Time To Live in seconds for cache objects created by this client. It is
-                possible to override this setting when calling the set method.
-            request_timeout_ms (Optional[int], optional): An optional timeout in milliseconds to allow for Get and Set
-                operations to complete. The request will be terminated if it takes longer than this value and will
-                result in TimeoutError. Defaults to None, in which case a 5 second timeout is used.
+            configuration (Configuration): An object holding configuration settings for communication with the server.
+            credential_provider (CredentialProvider): An object holding the auth token and endpoint information.
+            default_ttl (timedelta): A default Time To Live timedelta for cache objects created by this client.
+                It is possible to override this setting when calling the set method.
         Raises:
             IllegalArgumentError: If method arguments fail validations.
         """
-        _validate_request_timeout(request_timeout_ms)
-
-        endpoints = _momento_endpoint_resolver.resolve(auth_token)
-        self._control_client = _ScsControlClient(auth_token, endpoints.control_endpoint)
+        _validate_request_timeout(configuration.get_transport_strategy().get_grpc_configuration().get_deadline())
```

Comment:
Should this validation happen in the GRPC config object?

Reply:
Not a bad idea. I added validation in `StaticGrpcConfiguration`'s `withClientTimeout()`. I'm going to leave this here as a defensive measure though.